### PR TITLE
Refactors camera following into cameraFollowCoordinator; follow videos and analytics; new menu structure

### PR DESCRIPTION
**Pull request by benptc**
- cameraFollowCoordinator keeps track of all cameraFollowTargets equally - whether they are live virtualizers, recorded videos, or analytics poses; based on the selected followTarget, it will tell virtual camera to follow that target's sceneNode, and that's all the virtual camera needs to know!
- Adds new "Follow" menu to menu bar with items that get added for each new follow target in the scene. Also cleans up some items from the "Camera" menu and puts them in a newly-enabled sub-dropdown-menu of the Follow menu.
- Cleans up outdated logic from VirtualCamera and desktopCamera related to following, isRendering2d, and the even-older (no longer used) object targeting dropdown
- Eliminates references to the legacy unity project from desktopCamera and desktopAdapter
  - I can bring these back if we want to keep backwards compatible but at this point I feel it's only cluttering the code and most likely won't be used

Goes with: https://github.com/ptcrealitylab/vuforia-spatial-toolbox-userinterface/pull/513

### CODE REVIEW

File: content_scripts/CameraVis.js [hunk 1]
```diff
@@ -332,10 +346,6 @@ export class CameraVis {
             this.historyMesh.setPoints(this.historyPoints);
         }
 
-        if (this.sceneGraphNode) {
```
Comment (hobinjk-ptc) [inline content_scripts/CameraVis.js:335]: I believe this section is related to the jitter elimination work I did

File: content_scripts/desktopAdapter.js [hunk 2]
```diff
@@ -471,29 +315,11 @@ window.DEBUG_DISABLE_DROPDOWNS = false;
     }
 
     /**
-     * Adds socket.io listeners for matrix streams and UDP messages necessary to setup editor without mobile environment
+     * Adds socket.io listeners for UDP messages necessary to setup editor without mobile environment
      * (e.g. object discovery)
      */
     function addSocketListeners() {
 
-        // @deprecated
```
Comment (hobinjk-ptc) [inline content_scripts/desktopAdapter.js:479]: Great to have all the dead code removed!

File: content_scripts/VirtualCamera.js [hunk 3]
```diff
@@ -760,7 +756,8 @@ import * as THREE from '../../thirdPartyCode/three/three.module.js';
                 return; // don't animate the matrix with an infinite level of precision, stop when it gets very close to destination
             }
 
-            let shouldSmoothCamera = !this.isRendering2DVideo() && !this.zoomOutTransition;
+            // disables smoothing while following, to provide a tighter sync with the followed element
+            let shouldSmoothCamera = !this.followingState.active && !this.zoomOutTransition;
```
Comment (hobinjk-ptc) [inline content_scripts/VirtualCamera.js:760]: In general I feel like smoothing in 3rd person perspectives looks better

File: content_scripts/CameraVis.js (hunk 1 above)
Comment (benptc) [reply content_scripts/CameraVis.js:335]: Thanks for the review! This is helpful. After taking a step back I've noticed how bad the jitter is too and I'll make sure to fix it.

File: content_scripts/VirtualCamera.js (hunk 3 above)
Comment (benptc) [reply content_scripts/VirtualCamera.js:760]: Good point, I'll bring that back

**Review by hobinjk-ptc — APPROVED** [review]
LGTM with changes, great work!

**Pull request merged**